docs/proposals/rescheduling/rescheduling.md

```diff
@@ -0,0 +1,312 @@
+
+Supposing I deploy an application which depends on cluster infrastructure. Due to the sudden failure of the storage of member1, the application located in member1 suddenly fails.
+However, member1 is still available in Karmada'view and other applications which do not depend on storage work well.
+At this time, I want to be able to failover this single application.
+
+### Dependency
+
+Karmada's `Resource Interpreter Framework` is designed for interpreting resource structure. It provides users with a interpreter operation to tell Karmada
+how to figure out the health status of a specific object. It is up to users to decide when to reschedule.
+
+For example:
+
+```yaml
+apiVersion: config.karmada.io/v1alpha1
+kind: ResourceInterpreterCustomization
+metadata:
+  name: declarative-configuration-example
+spec:
+  target:
+    apiVersion: apps/v1
+    kind: Deployment
+  healthInterpretation:
+    luaScript: >
+      function InterpretHealth(observedObj)
+        return observedObj.status.readyReplicas == observedObj.spec.replicas
+      end
+```
+
+### Risks and Mitigations
+
+This proposal maintains the backward compatibility, the system built with previous versions of Karmada can be seamlessly migrated to the new version.
+The previous configurations(yamls) could be applied to the new version of Karmada and without any behavior change.
+
+## Design Details
+
+### API change
+
+Add related rescheduling configuration:
+- Threshold for how soon the unhealthy state of applications can be tolerated.
+    * Custom Configuration for specific resource types
+- Threshold for how soon a cluster can be rescheduled after being evicted.
+    * Custom Configuration for specific clusters
+- Threshold to protect slow starting containers. Sometimes, you have to deal with legacy applications that might
+require an additional startup time on their first initialization. In such cases, we need a parameter to cover the worse case startup time.
+    * Custom Configuration for specific resource types
+- Flag for whether to reschedule when the application is unhealthy
+    * Custom Configuration for specific resource types
+
+#### PropagationPolicy API change
+
+```golang
+type PropagationSpec struct {
+    ...
+    // ReschedulingTriggers represents the rescheduling triggers for unhealthy application state.
+    // +optional
+    ReschedulingTriggers []ReschedulingTrigger `json:"reschedulingTriggers,omitempty"`
+    ...
+}
+
+// ReschedulingTrigger represents the rescheduling trigger for matched resource objects.
+type ReschedulingTrigger struct {
+    // Index represents the corresponding index of ResourceSelectors.
+    // If this value is not set, it means that all resources hit by ResourceSectors use the same configuration.
+    // If the resource meets the conditions of multiple ResourceSelectors at the same time,
+    // it will use the matching configuration with the highest priority.
+    // +optional
+    Index *int32 `json:"index,omitempty"`
```

Comment:
Can we mimic the `downward API`?

First, we could rename the field to `ResourceSelectorsPath  json:"resourceSelectorsPath,omitempty"`,

and then allow the user to set it like this `resourceSelectorsPath: .spec.resourceSelectors` or `resourceSelectorsPath: .spec.resourceSelectors[0]`. 

`resourceSelectorsPath: .spec. resourceSelectors`  means that all resources hit by ResourceSectors use the same configuration.